device/device.go

```diff
@@ -35,8 +40,27 @@ type Device struct {
 	m                mqtt.Client
 	interfaces       map[string]interfaces.AstarteInterface
 	astarteAPIClient *client.Client
-	// AutoReconnect sets whether the device should reconnect automatically
+	brokerURL        string
+	// AutoReconnect sets whether the device should reconnect automatically. Defaults to false.
```

Comment:
I'd be more explicit to underline the difference with `ConnectRetry` and write:

[...] sets whether the device should reconnect automatically if it loses the connection after establishing it. [...]